src/NBitcoin/BlockStake.cs

```diff
@@ -34,9 +34,9 @@ public BlockStake()
         {
         }
 
-        public BlockStake(byte[] bytes)
+        public BlockStake(Network network, byte[] bytes)
```

Comment:
We should probably not do that, ther eshould be a Load() method instead.

Reply:
OK yeah I have created a `BlockStake.Load` that takes in either `byte[]` and `Block`.

Reply:
why Load?

Reply:
@bokobza This is to keep the pattern consistent throughout. We load blocks and block headers like this. Also, it lessens the footprint that the `Network` class has in NBitcoin. We can rename it to something else?